src/main/java/com/jcabi/http/wire/AbstractHeaderBasedCachingWire.java

```diff
@@ -92,12 +92,14 @@ public final Response send(
         final InputStream content, final int connect, final int read
     ) throws IOException {
         final Response rsp;
-        if (method.equals(Request.GET)) {
-            rsp = this.consultCache(
+        if (!method.equals(Request.GET)
+            || (method.equals(Request.GET) && this.requestHasCmcHeader(headers))
```

Comment:
@amihaiemil can't we just use one condition `method.equals(Request.GET) && !this.requestHasCmcHeader(headers)` and invert `if` with `else`?